### DISCUSSION TITLE
Ha e Virtualizzazione

### DISCUSSION THREAD

**Discussion by barbatrukko** [indomus/forum#2274] [HUB: Home Assistant]
Salve,
attualmente la mia istanza di HA è su RasPi3 Model B... comincia ad andarmi stretta anche perché essendo 32 bit non supporto Matter.
Attendevo che scendessero i prezzi dei RasPi4 ma ancora siamo in alto mare... volevo provare gli OrangePi ma comunque ad ora devi aspettare che arrivino da lontano e soprattutto vorrei vederli meglio sul campo. In più sempre gli orangePi 5 permetterebbero con l'hardware che hanno la virtualizzazione ma al momento credo solo con il virtualizzatore di Linux.
Allora mi sono detto, visto che ci sono diverse offerte, prendo un NUC Intel con un Jqualcosa.... spendo uguale ma ho molte volte tanto.
Inoltre vorrei mettere su tramite proxmox 2 VM, una con da solo HASSOS e uno con IoTStack con influxDb ed altro in modo da disaccoppiarli e sfruttare influxDB anche per altro.
Se in un secondo momento volessi acquistare la skyConnect, riesco a collegarla alla porta del NUC e farla vedere direttamente ad HASSOS?

**Comment by indomus** [comment]
> Salve, attualmente la mia istanza di HA è su RasPi3 Model B... comincia ad andarmi stretta anche perché essendo 32 bit non supporto Matter. Attendevo che scendessero i prezzi dei RasPi4 ma ancora siamo in alto mare... volevo provare gli OrangePi ma comunque ad ora devi aspettare che arrivino da lontano e soprattutto vorrei vederli meglio sul campo. In più sempre gli orangePi 5 permetterebbero con l'hardware che hanno la virtualizzazione ma al momento credo solo con il virtualizzatore di Linux. Allora mi sono detto, visto che ci sono diverse offerte, prendo un NUC Intel con un Jqualcosa.... spendo uguale ma ho molte volte tanto. Inoltre vorrei mettere su tramite proxmox 2 VM, una con da solo HASSOS e uno con IoTStack con influxDb ed altro in modo da disaccoppiarli e sfruttare influxDB anche per altro. Se in un secondo momento volessi acquistare la skyConnect, riesco a collegarla alla porta del NUC e farla vedere direttamente ad HASSOS?

Al di là del fatto che raspberry ora è moribondo (piuttosto per un upgrade ora come ora buttati su [Mini PC](https://indomus.it/componenti/mini-pc-intel-nuc-e-similari/)). Intel NUC è la stessa cosa ma lo paghi il quadruplo solo perché è certificato Intel.
Quale che sia la tua scelta, per lo scenario che hai descritto la cosa migliore è docker, sul quale implementare Core (al massimo Supervised, sigh), Influx e quel che ti pare.

**Reply by barbatrukko** [reply]
Ciao,
si errore mio, nel mio gergo con NUC non indento la linea NUC di intel,  intendo mini pc NON arm (quindi sia intel che AMD, che ormai se ne trovano tantissimi in giro). Avevo puntato un N5105.
Però preferivo virtualizzare piuttosto che dokerizzare. Già IoT stack è dokerizzato, percui metterei su una VM IoTstak che a sua volta con doker mi da tutta una serie di cose senza sbattimenti, e su un'altra VM HASSOS per conto suo, che va per la sua strada a prescindere.

**Reply by snakuzzo** [reply]
premesso che ognuno fa come gli pare, ma non capisco quale sia il problema legato al dockerizzare tutto.

**Comment by barbatrukko** [comment]
perché IoT stack di per se ha le sue configurazioni multi Docker che vanno bene fra di loro e sono preconfigurate, HA preferisco la versione con HASSOS....

Altrimenti mi devo creare io un Docker compose unico....un unico file yaml....

**Reply by snakuzzo** [reply]
...come fanno in tantissimi, direi.

**Reply by barbatrukko** [reply]
Alla fine IoT Stack è un Docker-compose grande, fatto e mantenuto da altri, che ti consente di non preoccuparti di nulla (soprattutto nell'andare avanti dei vari componenti). C'è un file docker compose, più o meno testato, che comprende i componenti principali (influx, rednode, graphana, etc) già collaudato in qualche guida o usato da qualcuno di voi?
Al giorno d'oggi la risorsa più preziosa è il tempo, e ti assicuro che per quanto mi piaccia smanettare su queste cose, più vado avanti più mi manca il tempo per sbattere la testa, rifare configurazioni, ripristinare ambienti,  e più mi servono soluzioni testate ed affidabili.

**Reply by snakuzzo** [reply]
ma pure se non volessi toccare quel `docker-compose.yaml`, ne puoi creare un altro tuo...ci vogliono 30s
